climin/stops/stops.py

```diff
@@ -193,6 +194,16 @@ def __call__(self, info):
         return info['n_iter'] > self.after and info[self.key] >= self.minimal
 
 
+class IsNaN(object):
+    """Stop criterion that returns True if any value corresponding to user-specified keys is NaN."""
+
+    def __init__(self, keys=[]):
+        self.keys = keys
+
+    def __call__(self, info):
+        return any([np.isnan(info[key]) for key in self.keys])
```

Comment:
It would indeed generalize this stopping criterion. However, I think checking non-finiteness is very dangerous: Often in optimization problems, the default value is inf or -inf. So I would stick to the more specific version.